http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -456,6 +459,14 @@ protected AuthChallenge resolveCode(String code) {
             log.error("Stale token");
             return challenge(HttpStatus.SC_FORBIDDEN, AuthenticationError.Reason.STALE_TOKEN, null);
         }
+
+        String stateCookieValue = getCookieValue(deployment.getStateCookieName());
+        String nonceValue = token.getClaimValueAsString(NONCE);
+        if (nonceValue == null || !nonceValue.equals(String.valueOf(stateCookieValue.hashCode()))) {
+            log.error("Invalid nonce");
```

Comment:
I think it would be better to validate the nonce in the `TokenValidator#parseAndVerifyToken` method here:

https://github.com/wildfly-security/wildfly-elytron/blob/2.x/http/oidc/src/main/java/org/wildfly/security/http/oidc/TokenValidator.java#L85

Reply:
Its more straight forward to take the value from idToken which has been validated